src/runtime/HalideRuntime.h

```diff
@@ -1485,6 +1485,76 @@ struct halide_filter_metadata_t {
  * the -profile target flag, which runs a sampling profiler thread
  * alongside the pipeline. */
 
+/** A single hardware counter. A cpu can only track so many
+ * counters at once, so the OS time-multiplexes them. Each counter
+ * includes a value, the number of cycles the counter was enabled,
+ * and the number of cycles the counter was actually tracked. We
+ * can estimate the true count with ((double)count * enabled /
+ * running) */
+struct halide_profiler_hardware_counter {
+    uint64_t count, enabled, running;
+};
+
+typedef enum halide_profiler_hardware_counter_id_t {
```

Comment:
Do these enum values correspond to values from x86 specifications? (If so, should comment to that effect.)

Reply:
No, just a list of every reasonable possibility I could find in perf_event.h